src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/RijndaelManagedTransform.Browser.cs

```diff
@@ -0,0 +1,1282 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+
+namespace System.Security.Cryptography
+{
+    internal sealed class RijndaelManagedTransform : ICryptoTransform, ILiteSymmetricCipher
+    {
+        public const int BlockSizeBytes = 16; // 128 bits
+        private const int BlockSizeInts = BlockSizeBytes / 4;
+
+        private readonly bool _encrypting;
+
+        private int[] _encryptKeyExpansion;
+        private int[] _decryptKeyExpansion;
+
+        private readonly int _Nr;
+        private readonly int _Nk;
+
+        private readonly int[] _encryptindex;
+        private readonly int[] _decryptindex;
+
+        private int[] _IV;
+        private int[] _lastBlockBuffer;
+        private byte[]? _depadBuffer;
+
+        public RijndaelManagedTransform(ReadOnlySpan<byte> key,
+                                        ReadOnlySpan<byte> iv,
+                                        bool encrypting)
+        {
+            _encrypting = encrypting;
+            _Nr = GetNumberOfRounds(key);
+            _Nk = key.Length / 4;
+
+            // Precompute the modulus operations: these are performance killers when called frequently
+            _encryptindex = new int[BlockSizeInts * 3];
+            _decryptindex = new int[BlockSizeInts * 3];
+
+            Span<int> encryptindex1 = _encryptindex.AsSpan(0, BlockSizeInts);
+            Span<int> encryptindex2 = _encryptindex.AsSpan(BlockSizeInts, BlockSizeInts);
+            Span<int> encryptindex3 = _encryptindex.AsSpan(BlockSizeInts * 2, BlockSizeInts);
+
+            Span<int> decryptindex1 = _decryptindex.AsSpan(0, BlockSizeInts);
+            Span<int> decryptindex2 = _decryptindex.AsSpan(BlockSizeInts, BlockSizeInts);
+            Span<int> decryptindex3 = _decryptindex.AsSpan(BlockSizeInts * 2, BlockSizeInts);
+
+            for (int j = 0; j < BlockSizeInts; j++)
+            {
+                encryptindex1[j] = (j + 1) % BlockSizeInts;
+                encryptindex2[j] = (j + 2) % BlockSizeInts;
+                encryptindex3[j] = (j + 3) % BlockSizeInts;
+                decryptindex1[j] = (j - 1 + BlockSizeInts) % BlockSizeInts;
+                decryptindex2[j] = (j - 2 + BlockSizeInts) % BlockSizeInts;
+                decryptindex3[j] = (j - 3 + BlockSizeInts) % BlockSizeInts;
+            }
+
+            // we only support CBC mode
+            if (iv.Length / 4 != BlockSizeInts)
+                throw new CryptographicException(SR.Cryptography_InvalidIVSize);
```

Comment:
I think this would read more naturally as `iv.Length != BlockSizeBytes`.

The comment is somewhat misleading - whether ECB or CBC, we still only want to support AES block sizes. I would change the comment to something like "we only support the standard AES block size".